src/php/Analyzer.php

```diff
@@ -99,68 +97,78 @@ public function analyze($x, ?NodeEnvironment $nodeEnvironment = null): Node {
             return $this->analyzeArray($x, $nodeEnvironment);
-        } else if ($x instanceof Table) {
+        }
+
+        if ($x instanceof Table) {
             return $this->analyzeTable($x, $nodeEnvironment);
-        } else if ($x instanceof Tuple) {
-            if ($x[0] instanceof Symbol) {
-                switch ($x[0]->getName()) {
-                    case 'def':
-                        return $this->analyzeDef($x, $nodeEnvironment);
-                    case 'ns':
-                        return $this->analyzeNs($x, $nodeEnvironment);
-                    case 'fn':
-                        return $this->analyzeFn($x, $nodeEnvironment);
-                    case 'quote':
-                        return $this->analyzeQuote($x, $nodeEnvironment);
-                    case 'do':
-                        return $this->analyzeDo($x, $nodeEnvironment);
-                    case 'if':
-                        return $this->analyzeIf($x, $nodeEnvironment);
-                    case 'apply':
-                        return $this->analyzeApply($x, $nodeEnvironment);
-                    case 'let':
-                        return $this->analyzeLet($x, $nodeEnvironment);
-                    case 'php/new':
-                        return $this->analyzePhpNew($x, $nodeEnvironment);
-                    case 'php/->':
-                        return $this->analyzePhpObjectCall($x, $nodeEnvironment, false);
-                    case 'php/::':
-                        return $this->analyzePhpObjectCall($x, $nodeEnvironment, true);
-                    case 'php/aget':
-                        return $this->analyzePhpAGet($x, $nodeEnvironment);
-                    case 'php/aset':
-                        return $this->analyzePhpASet($x, $nodeEnvironment);
-                    case 'php/apush':
-                        return $this->analyzePhpAPush($x, $nodeEnvironment);
-                    case 'php/aunset':
-                        return $this->analyzePhpAUnset($x, $nodeEnvironment);
-                    case 'recur':
-                        return $this->analyzeRecur($x, $nodeEnvironment);
-                    case 'try':
-                        return $this->analyzeTry($x, $nodeEnvironment);
-                    case 'throw':
-                        return $this->analyzeThrow($x, $nodeEnvironment);
-                    case 'loop':
-                        return $this->analyzeLoop($x, $nodeEnvironment);
-                    case 'foreach':
-                        return $this->analyzeForeach($x, $nodeEnvironment);
-                    case 'defstruct*':
-                        return $this->analyzeDefStruct($x, $nodeEnvironment);
-                    default:
-                        return $this->analyzeInvoke($x, $nodeEnvironment);
-                }
-            } else {
-                return $this->analyzeInvoke($x, $nodeEnvironment);
-            }
-        } else {
+        }
+
+        if (!$x instanceof Tuple) {
```

Comment:
This is the only case I would not split. It think it is much easier to understand if you have it in the following form
```php
<?php
if ($x instanceof Tuple) { // Matches the style of the ifs above
  if ($x[0] instanceof Symbol) {
    switch ... 
  }

  return $this->analyzeInvoke($x, $nodeEnvironment);
}

// This Exception is now at the end and it's clear that a case above is missing
// If the exception is somewhere in between it is much harder to understand the logic of the code.
throw new AnalyzerException('Unhandled type: ' . var_export($x, true), null, null);
```

Reply:
achso! ok, got it!